snek.vy

```diff
@@ -26,28 +26,34 @@
 interface Multifab:
     def cache(code :Bytes[20000000]) -> bytes32:  # max size, typed as `bytes` in ABI
         nonpayable
-    def build(hash :bytes32) -> address:
+    def build(hash :bytes32, args :Bytes[20000000]) -> address:
         nonpayable
 
+fab: Multifab
+types: public(HashMap[String[32], bytes32])
+objects: public(HashMap[String[32], address])
 
 @external
 def __init__(fab :address):
     """ fab is a multifab initialized with types from this snek project """
-    pass
+    self.fab = Multifab(fab)
 
 @external
-def _bind(typename :String[32], hash :bytes32):
+def bind(typename :String[32], _hash :bytes32):
```

Comment:
pls put back the underscore as it signifies that it is a thing for accessing guts you don't normally need